models/user.model.js

```diff
@@ -1,8 +1,95 @@
 import mongoose, { Schema } from "mongoose";
+import bcrypt from "bcryptjs";
 
+/**
+ * A Mongoose schema representing a user account.
+ */
 const UserSchema = new mongoose.Schema({
-  username: { type: String },
-  urlImg: { type: String },
+  /**
+   * The user's username. Must be unique and have a maximum length of 15 characters.
+   */
+  usuario: {
+    type: String,
+    required: true,
+    unique: true,
+    maxlength: 15,
+  },
+  /**
+   * Indicates whether the user account is currently blocked.
+   */
+  isBlocked: {
+    type: Boolean,
+  },
+  /**
+   * The date and time when the user account was blocked, if applicable.
+   */
+  fechaBloqueo: {
+    type: Date,
+  },
+  /**
+   * The number of unsuccessful login attempts made by the user.
+   */
+  numIntentos: {
+    type: Number,
+  },
+  /**
+   * The ID of the user's role in the system.
+   */
+  ptrRol: {
+    type: Schema.Types.ObjectId,
+    ref: "Rol",
+  },
+  /**
+   * The user's email address. Must be unique and match the format of a valid email address.
+   */
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+    validate: {
+      validator: function (v) {
+        return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(v);
+      },
+      message: (props) => `${props.value} is not a valid email address.`,
+    },
+  },
+  /**
+   * The user's password. Must meet the following requirements:
+   * - Be at least 8 characters long
+   * - Be no more than 15 characters long
+   * - Contain at least one uppercase letter
+   * - Contain at least one lowercase letter
+   * - Contain at least one number
+   * - Contain at least one special character from the following set: !@#$%^&*()_+}{":;?/>.<,|=-
+   */
+  password: {
+    type: String,
+    required: true,
+    validate: {
+      validator: function (v) {
+        return /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*()_+}{"':;?/>.<,|=-])[a-zA-Z\d!@#$%^&*()_+}{"':;?/>.<,|=-]{8,15}$/.test(
+          v
+        );
+      },
+      message: () =>
+        `The password must be at least 8 characters long, no more than 15 characters long, and contain at least one uppercase letter, one lowercase letter, one number, and one special character from the following set: !@#$%^&*()_+}{":;?/>.<,|=-`,
+    },
+  },
+});
+
+/**
+ * A pre-save hook that hashes the user's password before saving it to the database.
```

Comment:
Aquí debe de haber un salto de línea:
A pre-save hook that hashes the user's password before saving it to the database.

@param {function} next - The callback function to be called after the hook completes.
@returns {void}